File: dev/front-end/vpl-fe-web-admin/src/app/features/evaluation/models/evaluation.ts
export class Evaluation {
    id!: number;
    year!: number;
    value!: number;
    vehicleId!: number;
    vehicleName?: string;
    referenceYearName!: number;
    referenceYearId!: number;

    // Propriedades de visualização
    isRemoving?: boolean;
}

File: dev/front-end/vpl-fe-app-public/lib/features/evaluations/models/evaluation.dart
import 'package:vpl/features/referenceYears/enums/price_reference.dart';

class Evaluation {
  late int id;
  late int referenceYearId;
  late int year;
  late double value;
  late int vehicleId;
  late String vehicleName;
  late int referenceYearName;
  late PriceReference priceReference;

  Evaluation.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    referenceYearId = json['referenceYearId'];
    year = json['year'];
    value = json['value'];
    vehicleId = json['vehicleId'];
    vehicleName = json['vehicleName'];
    referenceYearName = json['referenceYearName'];
    priceReference = PriceReference.values[json['referenceYearName']];
  }
}
